### PR TITLE
Bug fixes to bootstrap task + splitting out dotfiles portion

### PR DESCRIPTION
**Pull request by oniofchaos**
I decided to make the `dotfiles:bootstrap` task be simply about setting up the base libraries, code, and gems required to get started with development. I've created two new tasks, `dotfiles:symlinks` and `dotfiles:plugins` for the "non-essential" portions, like installing zsh or vundle. 

I've tested this a bunch on a Mac VM I created, and it should work on a fresh machine. The main issue I ran into was installing the global gems, but I think that's from permissions issues on the VM (I hope).

### CODE REVIEW

File: rake_helpers/ruby.rb [hunk 1]
```diff
@@ -31,7 +35,7 @@ def use_latest_ruby
     end
 
     def install_global_gems
-      puts 'installing global gems'
+      puts 'Installing global gems'
```
Comment (shicholas) [inline rake_helpers/ruby.rb]: given that we now have rvm here, I think we can install in [the rvm global gemset](https://rvm.io/gemsets/global). Though tbh, these gems are personal and may be better off being excluded from this script.

File: rake_helpers/homebrew.rb [hunk 2]
```diff
@@ -21,51 +19,51 @@ def install_or_update_homebrew
...
+
+      system('brew doctor')
+
+      install_packages
     end
 
     def install_packages
       puts 'Installing packages'
 
       `brew install \
```
Comment (oniofchaos) [inline rake_helpers/homebrew.rb]: @shicholas the install step here takes a really long time. Are there any optional packages that you would recommend for removal?

Comment (shicholas) [inline rake_helpers/homebrew.rb]: I think every package that's not needed for our rails app can be optional. So everything except men cached, Postgres, etc.

Comment (oniofchaos) [inline rake_helpers/homebrew.rb]: Awesome. That brought it down from really slow to just slow.

**Comment by oniofchaos** [comment]
I've tested this a bunch and it appears to work fine, at least on my experimental VM. We can give it a live test next week when Bijal starts and go from there.

File: rake_helpers/homebrew.rb (hunk 2 above)
Comment (shicholas) [inline rake_helpers/homebrew.rb]: :+1:

**Comment by shicholas** [comment]
:+1:

**Pull request merged**